cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java

```diff
@@ -325,18 +325,19 @@ int getPartition(ScheduleId scheduleId) {
   }
 
   @Override
-  public String retrieveSubscriberState(String topic) {
-    Row row = table.get(getRowKey(topic));
+  public String retrieveSubscriberState(String topic, String subscriber) {
+    Row row = table.get(getRowKey(topic, subscriber));
     byte[] messageIdBytes = row.get(COL);
     return messageIdBytes == null ? null : Bytes.toString(messageIdBytes);
   }
 
   @Override
-  public void persistSubscriberState(String topic, String messageId) {
-    table.put(getRowKey(topic), COL, Bytes.toBytes(messageId));
+  public void persistSubscriberState(String topic, String subscriber, String messageId) {
+    table.put(getRowKey(topic, subscriber), COL, Bytes.toBytes(messageId));
   }
 
-  private byte[] getRowKey(String topic) {
+  private byte[] getRowKey(String topic, String subscriber) {
```

Comment:
why adding this method if the subscriber is not used?

Reply:
It is intentional for future expansion such that there is one method to modify instead of two

Reply:
on a second thought, this class might not be changed. Removing the subscriber parameter.